ckanext/activity/tests/logic/test_action.py

```diff
@@ -840,6 +948,78 @@ def test_limit_hits_max(self):
         results = helpers.call_action("user_activity_list", id=id, limit="9")
         assert len(results) == 7  # i.e. ckan.activity_list_limit_max
 
+    def test_private_dataset_activities_not_visible_to_users_without_permission(self):
+        user = factories.User()
+        auth_user_obj = model.User.get(user["id"])
+        another_user = factories.User()
+        org = factories.Organization(user=another_user)
+        _clear_activities()
+        dataset = factories.Dataset(owner_org=org["id"], private=True)
+
+        dataset["private"] = False
+        helpers.call_action(
+            "package_update", context={"user": another_user["name"]}, **dataset
+        )
+        activities = helpers.call_action(
+            "user_activity_list",
+            id=another_user["id"],
+            context={"user": user["name"], "auth_user_obj": auth_user_obj}
+        )
+
+        assert len(activities) == 1
+        assert activities[0]["activity_type"] == "changed package"
```

Comment:
These sorts of asserts tell us more when they fail than the `len()` asserts
```suggestion
        assert [activity['activity_type'] for activity in activities] == ['changed package']
```